src/Common/Bridge/BridgeFacadeMethodsRule.php

```diff
@@ -0,0 +1,192 @@
+            !$node instanceof ClassNode
+        ) {
+            return;
+        }
+        $bridgeName = $node->getFullQualifiedName();
+
+        foreach ($node->getMethods() as $method) {
+            $interfaceMethodName = sprintf('%s::%s', $bridgeName, $method->getName());
+            $interfaceMethodReflection = new ReflectionMethod($interfaceMethodName);
+            $methodReturnType = $interfaceMethodReflection->getReturnType();
+
+            if (preg_match('/^(get|read|find)/', $method->getName())) {
+                $this->verifyGetMethod($method, $methodReturnType);
+
+                continue;
+            }
+
+            if (preg_match('/^(delete|remove)/', $method->getName())) {
+                $this->verifyDeleteMethod($method, $methodReturnType);
+
+                continue;
+            }
+
+            if (preg_match('/^(create|add)/', $method->getName())) {
+                $this->verifyCreateMethod($method, $methodReturnType);
+
+                continue;
+            }
+
+            if (preg_match('/^(update|change)/', $method->getName())) {
+                $this->verifyUpdateMethod($method, $methodReturnType);
+
+                continue;
+            }
+
+            if (strpos($method->getName(), 'save') === 0) {
+                $this->addViolation($method, [sprintf(
+                    'Method `%s()` must have `public function update(create)<DomainEntity>Collection(<DomainEntity>CollectionRequestTransfer): <DomainEntity>CollectionResponseTransfer;` signature.',
+                    $method->getName(),
+                )]);
+
+                continue;
+            }
+        }
+    }
+
+    /**
+     * @param \PHPMD\Node\MethodNode $method
+     * @param \ReflectionType|null $methodReturnType
+     *
+     * @return void
+     */
+    protected function verifyGetMethod(MethodNode $method, ?ReflectionType $methodReturnType): void
+    {
+        $parameters = $method->getNode()->getParameters();
+
+        if (
+            $method->getParameterCount() !== 1 ||
+            !$parameters[0]->getClass() ||
+            !$methodReturnType ||
+            preg_match('/^get(?<domainEntity>\w+)Collection$/', $method->getName(), $methodNameMatches) === 0 ||
+            preg_match('/^(?<domainEntity>\w+)CriteriaTransfer$/', $parameters[0]->getClass()->getName(), $methodParameterMatches) === 0 ||
+            preg_match('/^Generated\\\\Shared\\\\Transfer\\\\(?<domainEntity>\w+)CollectionTransfer$/', $methodReturnType->getName(), $methodReturnTypeMatches) === 0 ||
+            count(array_unique([$methodNameMatches['domainEntity'], $methodParameterMatches['domainEntity'], $methodReturnTypeMatches['domainEntity']])) !== 1
+        ) {
+            $this->addViolation($method, [sprintf(
+                'Method `%s()` must have `public function get<DomainEntity>Collection(<DomainEntity>CriteriaTransfer): <DomainEntity>CollectionTransfer;` signature.',
```

Comment:
; should be removed as they are not part of the signature (depending on interface vs method).